aip/general/0236.md

```diff
@@ -0,0 +1,321 @@
+be the live policy. This mechanism allows API consumer to observe changes
+proposed by an experiment without risking unintended side effects for live
+traffic.
+
+### Non-goals
+
+#### Non-policy resources
+
+This proposal is for a safety mechanism for policy rollouts only. Safe rollouts
+for resources that are not governed by "policies" are not in scope.
+
+A policy is a resource that provides rules that admit or deny access to other
+resources. Generally, the outcome of a policy can be evaluated to a specific set
+of outcomes. Firewall policies exemplify a case that is suitable for preview.
+The system simply logs the traffic that the firewall policy would accept or
+deny before making live changes to the configuration.
+
+### Nested collection called experiments
+
+A new configuration of a policy to be validated using preview mode is stored as
+a nested collection under that policy resource.
+
+A hypothetical policy Resource called, `Policy`, will be used throughout. It
+has the following resource name pattern:
+
+`projects/{project}/locations/{location}/policies/{policy}`
+
+The experimental versions of the resource used for preview mode are represented
+as a nested collection under `Policy`. The resource type will follow the
+convention "<resource_name> + Experiment”. The following is the pattern for the
+experiment:
+
+`projects/{project}/locations/{location}/policies/{policy}/experiments/{experiment}`
+
+A proto used to represent an experiment **must** contain the following fields:
+
+  1. The required top-level fields for a resource, like `name` and `etag`
+  2. The policy that is being tested itself
+  3. The field, `preview_metadata` for preview
+
+
+```proto
+message PolicyExperiment {
+
+  // The name of the PolicyExperiment.
+  string name = 1;
+ 
+  // The policy experiment. This Policy will be used to simulate the effects of
+  // the change but will affect life traffic.
+  Policy policy = 2;
+
+  // The metadata associated with this policy experiment.
+  PreviewMetadata preview_metadata = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+}
+```
+- A new proto **must** be used for each experiment. There **must not** be a
+  common <ResourceName>Preview proto for all experiments.
+  - It **must not** use the same resource type as the live policy.
+- The experiment proto **must** contain a top-level field with the same type as
+   the live policy. 
+  - It **must** be named as the live resource type. For example, if the
+    experiment is for FirewallPolicy, then this field **must** be named
+    firewall_policy.
+- The name inside the embedded policy message **must** be the name of the live
+  policy. 
+- When the user is ready to promote an experiment, they **must** copy the 
+  `policy` message into the live policy and delete the experiment. This can be
+  done manually or via a "commit" method.
+- A product **may** support multiple simulated configurations concurrently in
```

Comment:
This is where I'd refer to "multiple experiments" (to avoid introducing new terminology like "simulated configurations").